ntp.go

```diff
@@ -205,14 +243,30 @@ func getTime(host string, version int) (*msg, error) {
 		return nil, err
 	}
 
+	// It's possible to use random uint64 as client's `TransmitTime` field,
+	// it has better privacy (clock of the node is not disclosed in
+	// plain-text), better UDP packet spoofing resistance (blind attacker
+	// has to guess both port and the uint64 value), and OpenNTPD behaves
+	// like that. But math/rand is not secure enough for the purpose,
+	// crypto/rand takes 64 bits of entropy for every outgoing packet and
+	// CSPRNG from crypto/rand/rand_unix is not available: see
+	// https://github.com/golang/go/issues/13820
+	if m.OriginTime != xmt {
```

Comment:
There are some of them in [RFC5905](https://tools.ietf.org/html/rfc5905#section-8).

> A packet is bogus if the origin timestamp t1 in the packet does not match the xmt state variable T1.